lisa/sut_orchestrator/azure/common.py

```diff
@@ -1904,3 +1916,271 @@ def get_size(disk_type: schema.DiskType, data_disk_iops: int = 1) -> int:
             return iops_dict[min_iops]
         else:
             raise LisaException(f"Data disk type {disk_type} is unsupported.")
+
+
+def get_certificate_client(
+    vault_url: str, platform: "AzurePlatform"
+) -> CertificateClient:
+    return CertificateClient(vault_url, platform.credential)
+
+
+def get_secret_client(vault_url: str, platform: "AzurePlatform") -> SecretClient:
+    return SecretClient(vault_url, platform.credential)
+
+
+def get_key_vault_management_client(
+    platform: "AzurePlatform",
+) -> KeyVaultManagementClient:
+    return KeyVaultManagementClient(platform.credential, platform.subscription_id)
+
+
+def get_tenant_id(credential: Any) -> Any:
+    # Initialize the Subscription client
+    subscription_client = SubscriptionClient(credential)
+    # Get the subscription
+    subscription = next(subscription_client.subscriptions.list())
+    return subscription.tenant_id
+
+
+def get_identity_id() -> Any:
+    # Define constants
+    graph_api_url = "https://graph.microsoft.com/.default"
+    request_url = "https://graph.microsoft.com/v1.0/me"
+
+    # Get a token for the Microsoft Graph API
+    token_credential = DefaultAzureCredential()
+    token = token_credential.get_token(graph_api_url)
+
+    # Set up the API call headers
+    headers = {
+        "Authorization": f"Bearer {token.token}",
+        "Content-Type": "application/json",
+    }
+
+    # Set a timeout of 10 seconds for the request
+    response = requests.get(request_url, headers=headers, timeout=10)
```

Comment:
in our pipeline we use app token not user token, so we will encounter below error, we should handle both situation.
```
Failed to retrieve user object ID. Status code: 400. Response: {"error":{"code":"BadRequest","message":"/me request is only valid with delegated authentication flow.","innerError":{"date":"2023-08-17T02:25:00","request-id":"baede3c6-8aa1-46da-9c64-d56a4a751d8a","client-request-id":"baede3c6-8aa1-46da-9c64-d56a4a751d8a"
```

Reply:
Ok I will start some research.